### PR TITLE
libxslt: build with crypto

### PR DESCRIPTION
**Pull request by SeekingMeaning**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

License PR: https://github.com/Homebrew/homebrew-core/pull/67205

### CODE REVIEW

File: Formula/libxslt.rb [hunk 1]
```diff
@@ -32,11 +32,17 @@ class Libxslt < Formula
...
+      "--disable-dependency-tracking",
+      "--disable-silent-rules",
+      "--prefix=#{prefix}",
+      "--without-python",
+      "--with-libxml-prefix=#{Formula["libxml2"].opt_prefix}",
+    ]
+    on_linux do
+      args << "--without-crypto"
+    end
+    system "./configure", args
```
Comment (fxcoudert) [inline Formula/libxslt.rb:45]:
```suggestion
    system "./configure", *args
```

**Review by fxcoudert — DISMISSED** [review]

**Comment by fxcoudert** [comment]
If we're adding linux-specific code to the formula, shouldn't the PR be run on linux CI?

**Comment by SeekingMeaning** [comment]
This actually isn't be needed as I managed to build this locally without the flag, which was added 6 years ago: https://github.com/Linuxbrew/legacy-linuxbrew/pull/147

Closing this in favor of https://github.com/Homebrew/linuxbrew-core/pull/21845

**Comment by SeekingMeaning** [comment]
Reopening this as `libgcrypt` could become a circular dependency without the `--without-crypto` flag

**Comment by iMichka** [comment]
I think the flags was added to avoid opportunistic linkage to libgcrypt.
On the other side we have a cyclic dependency here ... wondering why libxslt needs encryption.

**Comment by SeekingMeaning** [comment]
On Arch Linux, `libxslt`<sup>[1]</sup> depends on `libgcrypt`<sup>[2]</sup> but not vice versa

1. https://archlinux.org/packages/extra/x86_64/libxslt/
2. https://archlinux.org/packages/core/x86_64/libgcrypt/

**Comment by iMichka** [comment]
So I would be in favour of building `libgcrypt` without `libxslt`.

**Comment by SeekingMeaning** [comment]
Should we add `libgcrypt` as an explicit dependency for this formula?

**Comment by iMichka** [comment]
Yes, let's enable crypto for the formula, so we do the same as arch linux.

**Comment by iMichka** [comment]
This needs a revision bump as we are adding a new dependency. The rest looks good, thanks for adding an extended test.

**Review by dawidd6 — APPROVED** [review]

**Pull request closed without merging**